### PR TITLE
[improve][broker] Improve managed ledger factory async delete method parametres.

### PR DESCRIPTION
**Pull request by mattisonchao**
### Motivation

https://github.com/apache/pulsar/pull/17915 added a new public API for the managed ledger factory interface. it's not a good approach to let the user pass a future instead of the managed configuration.

### Modifications

Improve the method parameters and keep the compatibility.

### Verifying this change

- [x] Make sure that the change passes the CI checks.

### Documentation

- [ ] `doc` 
- [ ] `doc-required` 
- [x] `doc-not-needed` 
- [ ] `doc-complete`

### CODE REVIEW

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java [hunk 1]
```diff
@@ -25,12 +25,7 @@
 import com.google.common.collect.Maps;
 import io.netty.util.concurrent.DefaultThreadFactory;
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-import java.util.TreeMap;
-import java.util.UUID;
+import java.util.*;
```
Comment (Demogorgon314) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java:28]: nit: Please fix the code format

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java [hunk 2]
```diff
@@ -826,12 +824,39 @@ public void asyncDelete(String name, DeleteLedgerCallback callback, Object ctx)
...
+                            @Nonnull DeleteLedgerCallback callback, @Nullable Object ctx) {
+        Objects.requireNonNull(name);
+        Objects.requireNonNull(managedLedgerConfig);
+        Objects.requireNonNull(callback);
+
+        final CompletableFuture<ManagedLedgerImpl> future = ledgers.get(name);
         if (future == null) {
             // Managed ledger does not exist and we're not currently trying to open it
-            deleteManagedLedger(name, mlConfigFuture, callback, ctx);
+            deleteManagedLedger(name, managedLedgerConfig, callback, ctx);
```
Comment (heesung-sn) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/ManagedLedgerFactoryImpl.java:859]: Why not keep using `mlConfig` var name?

File: managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedLedgerFactory.java [hunk 3]
```diff
@@ -183,9 +185,24 @@ void delete(String name, CompletableFuture<ManagedLedgerConfig> mlConfigFuture)
      * @throws InterruptedException
      * @throws ManagedLedgerException
      */
+    @Deprecated
     void asyncDelete(String name, CompletableFuture<ManagedLedgerConfig> mlConfigFuture,
```
Comment (heesung-sn) [inline managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedLedgerFactory.java:189]: What's our plan to remove this deprecated code?  Can we clean the code to make the caller use the new one?

Comment (coderzc) [reply managed-ledger/src/main/java/org/apache/bookkeeper/mledger/ManagedLedgerFactory.java:189]: +1

**Pull request closed without merging**